Submitted By: Fernando de Oliveira <famobr at yahoo dot com dot br>
Date: 2013-07-11
Initial Package Version: 1.4 (though original post mentions head, 1.3, 1.2 as well)
Upstream Status: Submitted
Origin: Jiri Vanek <jvanek at redhat.com>
http://mail.openjdk.java.net/pipermail/distro-pkg-dev/2013-June/023745.html
Description: Latest OpenJDK breaks Java browser plugin
diff -Naur icedtea-web-1.4.orig/netx/net/sourceforge/jnlp/runtime/Boot.java icedtea-web-1.4/netx/net/sourceforge/jnlp/runtime/Boot.java
--- icedtea-web-1.4.orig/netx/net/sourceforge/jnlp/runtime/Boot.java	2013-05-03 16:08:24.274519818 -0300
+++ icedtea-web-1.4/netx/net/sourceforge/jnlp/runtime/Boot.java	2013-07-12 10:50:41.120910762 -0300
@@ -35,6 +35,8 @@
 import net.sourceforge.jnlp.cache.UpdatePolicy;
 import net.sourceforge.jnlp.security.viewer.CertificateViewer;
 import net.sourceforge.jnlp.services.ServiceUtil;
+import sun.awt.AppContext;
+import sun.awt.SunToolkit;
 
 /**
  * This is the main entry point for the JNLP client.  The main
@@ -113,6 +115,9 @@
      * Launch the JNLP file specified by the command-line arguments.
      */
     public static void main(String[] argsIn) {
+        if (AppContext.getAppContext() == null) {
+            SunToolkit.createNewAppContext();
+        }
         args = argsIn;
 
         if (null != getOption("-viewer")) {
diff -Naur icedtea-web-1.4.orig/netx/net/sourceforge/jnlp/runtime/JNLPRuntime.java icedtea-web-1.4/netx/net/sourceforge/jnlp/runtime/JNLPRuntime.java
--- icedtea-web-1.4.orig/netx/net/sourceforge/jnlp/runtime/JNLPRuntime.java	2013-05-03 16:08:24.280519745 -0300
+++ icedtea-web-1.4/netx/net/sourceforge/jnlp/runtime/JNLPRuntime.java	2013-07-12 10:55:13.751999632 -0300
@@ -234,7 +234,7 @@
         try {
             UIManager.setLookAndFeel(UIManager.getSystemLookAndFeelClassName());
         } catch (Exception e) {
-            // ignore it
+            e.printStackTrace();
         }
 
         doMainAppContextHacks();
diff -Naur icedtea-web-1.4.orig/plugin/icedteanp/java/sun/applet/PluginMain.java icedtea-web-1.4/plugin/icedteanp/java/sun/applet/PluginMain.java
--- icedtea-web-1.4.orig/plugin/icedteanp/java/sun/applet/PluginMain.java	2013-05-03 16:08:24.491517092 -0300
+++ icedtea-web-1.4/plugin/icedteanp/java/sun/applet/PluginMain.java	2013-07-12 10:59:10.906708012 -0300
@@ -72,6 +72,8 @@
 import java.net.ProxySelector;
 import java.util.Enumeration;
 import java.util.Properties;
+import sun.awt.AppContext;
+import sun.awt.SunToolkit;
 
 import net.sourceforge.jnlp.config.DeploymentConfiguration;
 import net.sourceforge.jnlp.runtime.JNLPRuntime;
@@ -94,6 +96,9 @@
      */
     public static void main(String args[])
             throws IOException {
+        if (AppContext.getAppContext() == null) {
+            SunToolkit.createNewAppContext();
+        }
         if (args.length != 2 || !(new File(args[0]).exists()) || !(new File(args[1]).exists())) {
             System.err.println("Invalid pipe names provided. Refusing to proceed.");
             System.exit(1);
